File: tests/fixtures/basic.php
<?php

const USERS_SQL = <<<'SQL'
      SELECT id, email
      FROM users
      WHERE status = 'active'
    SQL;

$summarySql = 'SELECT status, count(*) AS total ' . 'FROM users ' . 'GROUP BY status ' . 'HAVING count(*) > 0';

$deleteSql = 'DELETE FROM users ' . "WHERE status = 'inactive'";

$db = new class {
    public function query(string $sql, mixed ...$params): string
    {
        return $sql;
    }

    public function execute(string $sql, mixed ...$params): string
    {
        return $sql;
    }

    public function prepare(string $sql): string
    {
        return $sql;
    }
};

$rows = $db->execute(<<<'SQL'
      UPDATE users
      SET status = ?
      WHERE email = ?
    SQL, 'active', 'alice@example.com');

$inserts = $db->query(
    'INSERT INTO users (email, status) ' . 'VALUES (?, ?) ' . 'RETURNING id, email, status',
    'alice@example.com',
    'active',
);

$cte = $db->query(
    'WITH recent_users AS ( '
    . "SELECT id, email FROM users WHERE created_at >= NOW() - INTERVAL '7 days' "
    . ') '
    . 'SELECT id, email FROM recent_users '
    . 'ORDER BY email ASC',
);

$stmt = $db->prepare('CREATE TABLE audit_logs (id BIGINT PRIMARY KEY)');
$alter = $db->prepare(<<<'SQL'
      ALTER TABLE audit_logs
      ADD COLUMN created_at TIMESTAMP
    SQL);

$joinSql = <<<'SQL'
      SELECT u.id, u.email, p.name
      FROM users u
      LEFT JOIN projects p ON u.id = p.user_id
      WHERE u.id IN (SELECT user_id FROM audit_logs GROUP BY user_id HAVING COUNT(*) > 1)
      ORDER BY u.created_at
    SQL;

$windowSql = <<<'SQL'
      WITH ranked AS (
        SELECT id, email, row_number() OVER (PARTITION BY status ORDER BY created_at) AS rn
        FROM users
      )
      SELECT id, email FROM ranked WHERE rn <= 5
    SQL;

$truncateSql = <<<'SQL'
      TRUNCATE TABLE audit_logs
    SQL;

$dropSql = <<<'SQL'
      DROP TABLE IF EXISTS temp_projects
    SQL;

$unionSql = <<<'SQL'
      SELECT id, email FROM users WHERE status = 'active'
      UNION
      SELECT id, email FROM archived_users WHERE status = 'active'
    SQL;

$existsSql = <<<'SQL'
      SELECT id, email FROM users u
      WHERE EXISTS (SELECT 1 FROM projects p WHERE p.user_id = u.id)
    SQL;

$transactionSql = <<<'SQL'
      BEGIN;
      UPDATE accounts SET balance = balance - 100 WHERE id = 1;
      UPDATE accounts SET balance = balance + 100 WHERE id = 2;
      COMMIT;
    SQL;

$upsertSql = $db->query(
    'INSERT INTO users (email, status) '
    . 'VALUES (?, ?) '
    . 'ON CONFLICT (email) DO UPDATE SET status = excluded.status',
    'bob@example.com',
    'active',
);

File: tests/fixtures/basic.cs
using System;

class Db {
    public string Query(string sql, params object[] args) => sql;
    public string Execute(string sql, params object[] args) => sql;
    public string Prepare(string sql) => sql;
}

class Program {
    static void Main() {
        const string USERS_SQL = @"
SELECT Id, Email
FROM Users
WHERE Status = 'active'
";

        var summarySql = "SELECT Status, COUNT(*) AS Total " +
            "FROM Users " +
            "GROUP BY Status " +
            "HAVING COUNT(*) > 0";

        var deleteSql = "DELETE FROM Users " +
            "WHERE Status = 'inactive'";

        var db = new Db();

        var rows = db.Execute(@"
UPDATE Users
SET Status = @status
WHERE Email = @email
",
            "active",
            "alice@example.com"
        );

        var inserts = db.Query(
            "INSERT INTO Users (Email, Status) " +
            "VALUES (@email, @status) " +
            "RETURNING Id, Email, Status",
            "alice@example.com",
            "active"
        );

        var cte = db.Query(
            "WITH recent_users AS ( " +
            "SELECT Id, Email FROM Users WHERE CreatedAt >= NOW() - INTERVAL '7 days' " +
            ") " +
            "SELECT Id, Email FROM recent_users " +
            "ORDER BY Email ASC"
        );

        var stmt = db.Prepare("CREATE TABLE AuditLogs (Id BIGINT PRIMARY KEY)");
        var alter = db.Prepare(@"
ALTER TABLE AuditLogs
ADD COLUMN CreatedAt TIMESTAMP
");

        Console.WriteLine(USERS_SQL);
        Console.WriteLine(summarySql);
        Console.WriteLine(rows);
        Console.WriteLine(inserts);
        Console.WriteLine(cte);
        Console.WriteLine(stmt);
        Console.WriteLine(deleteSql);
        var joinSql = @"SELECT u.id, u.email, p.name
FROM users u
LEFT JOIN projects p ON u.id = p.user_id
WHERE u.id IN (SELECT user_id FROM audit_logs GROUP BY user_id HAVING COUNT(*) > 1)
ORDER BY u.created_at";

        var windowSql = @"WITH ranked AS (
    SELECT id, email, row_number() OVER (PARTITION BY status ORDER BY created_at) AS rn
    FROM users
)
SELECT id, email FROM ranked WHERE rn <= 5";

        Console.WriteLine(alter);
        Console.WriteLine(joinSql);
        Console.WriteLine(windowSql);
    }
}
